tests/permissions.spec.ts

```diff
@@ -0,0 +1,393 @@
+import { test, expect, LxdVersions } from "./fixtures/lxd-test";
+import {
+  toggleIdentitiesForGroups,
+  addPermission,
+  createGroup,
+  deleteGroup,
+  renameGroup,
+  randomGroupName,
+  removePermission,
+  restorePermission,
+  selectGroupAction,
+  visitGroupsPage,
+  selectGroupsToModify,
+} from "./helpers/auth-groups";
+import {
+  identityBar,
+  identityFoo,
+  selectIdentitiesToModify,
+  toggleGroupsForIdentities,
+  visitIdentitiesPage,
+} from "./helpers/auth-identities";
+import {
+  createIdpGroup,
+  deleteIdpGroup,
+  editIdpGroup,
+  randomIdpGroupName,
+  visitIdpGroupsPage,
+} from "./helpers/auth-idp-groups";
+import {
+  assertModificationStatus,
+  confirmGroupsModifiedForIdentity,
+  confirmIdentitiesModifiedForGroup,
+  redoChange,
+  undoChange,
+} from "./helpers/auth";
+import {
+  resetSetting,
+  updateSetting,
+  visitServerSettings,
+} from "./helpers/server";
+
+const skipIfNotSupported = (lxdVersion: LxdVersions) =>
+  test.skip(
+    lxdVersion === "5.0-edge",
+    "Fine grained authorisation is not available for lxd 5.0",
+  );
+
+test("LXD 5.0 does not support permissions", async ({ page, lxdVersion }) => {
+  test.skip(
+    lxdVersion !== "5.0-edge",
+    "Fine grained authorisation is supported in lxd 5.21 LTS and latest/edge",
+  );
+  await page.goto("/ui/");
+  await expect(page.getByRole("button", { name: "Permissions" })).toBeHidden();
+});
+
+test.describe("Permission groups", () => {
+  test("create and delete group", async ({ page, lxdVersion }) => {
+    skipIfNotSupported(lxdVersion);
+    const group = randomGroupName();
+    await visitGroupsPage(page);
+    await createGroup(page, group, `${group}-desc`);
+    await deleteGroup(page, group);
+  });
+
+  test("edit and rename group", async ({ page, lxdVersion }) => {
+    skipIfNotSupported(lxdVersion);
+    const group = randomGroupName();
+    const newGroupName = randomGroupName();
+    await createGroup(page, group, group);
+    await renameGroup(page, group, newGroupName);
+    await renameGroup(page, newGroupName, group);
+    await deleteGroup(page, group);
+  });
+
+  test("add new permissions to group", async ({ page, lxdVersion }) => {
+    skipIfNotSupported(lxdVersion);
+    const group = randomGroupName();
+    await createGroup(page, group, group);
+    await selectGroupAction(page, group, "Manage permissions");
+    await addPermission(page, "server", "server", "admin");
+    await removePermission(page, "server", "server", "admin");
+    await restorePermission(page, "server", "server", "admin");
+    await undoChange(page);
+    await expect(page.getByText("1 permission will be modified")).toBeHidden();
+    await redoChange(page);
+    await assertModificationStatus(page, "1 permission will be modified");
+    await page.getByRole("button", { name: "Save changes" }).click();
+    await page.waitForSelector(`text=Permissions for group ${group} updated.`);
```

Comment:
Maybe add a check that the group has 1 permission assigned? Currently we don't ensure the permission is saved correctly.